gms/gossiper.cc

```diff
@@ -675,15 +675,17 @@ future<> gossiper::remove_endpoint(inet_address endpoint, permit_id pid) {
         logger.warn("Fail to call on_remove callback: {}", std::current_exception());
     }
 
+    auto lock = co_await lock_endpoint_update_semaphore();
     if(_seeds.contains(endpoint)) {
         build_seeds_list();
         _seeds.erase(endpoint);
         logger.info("removed {} from _seeds, updated _seeds list = {}", endpoint, _seeds);
     }
 
-    _live_endpoints.erase(endpoint);
-    co_await update_live_endpoints_version();
-    _unreachable_endpoints.erase(endpoint);
+    co_await mutate_live_and_unreachable_endpoints([endpoint] (gossiper& g) {
```

Comment:
Aren't we now calling `lock_endpoint_update_semaphore()` inside `mutate_...` while the lock is already being held (we took it above)? Doesn't this deadlock?

Reply:
oops, sorry, the lock is removed in the next patch by mistake.  Likely due to a pilot error while editing the series.  I'll rebase and make sure it's gone at this stage.

Reply:
fixed in v5